### DISCUSSION TITLE
How to install clair and integration with harbor

### DISCUSSION THREAD

**Discussion by asd99557** [quay/clair#1395] [Q&A] [answered]
How to install clair scanner and integration with harbor

**Comment by hdonnay** [comment, marked as answer]
There's no explicit "installation" process for clair, it just needs a database and then to be configured to use it. I'd recommend using the container to run the service, but running a binary directly should be fine.

As far as integration with Harbor, what have you tried? I don't think Harbor speaks our API, so you'll probably need to implement support in Harbor or write an adapter service.

**Comment by asd99557** [comment]
Thanks for the quick response.

When I try integration with Harbor it's given unreachable error message.

So that reason I am asking how to integration Harbor with clair.

Any specific step like having confusion in git repo if you can help me with
that?

My Harbor setup is ready but I am not clear in clair integration.

**Reply by hdonnay** [reply]
I haven't tried to use Clair with Harbor; to my knowledge you're the first to try.
Like I said, the APIs are likely incompatible so you'll probably have to modify Harbor or write an adapter service.

If you have *specific* questions, I can help.

**Comment by asd99557** [comment]
Ok, thanks.

Do you have knowledge about how to modify Harbor or creat adapter service ?

**Reply by hdonnay** [reply]
No, I don't know anything about what API Harbor expects for its container security scanner.
